src/Rules/Properties/MissingReadOnlyByPhpDocPropertyAssignRule.php

```diff
@@ -0,0 +1,127 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Properties;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Node\ClassPropertiesNode;
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\ShouldNotHappenException;
+use ReflectionException;
+use function array_key_exists;
+use function explode;
+use function sprintf;
+
+/**
+ * @implements Rule<ClassPropertiesNode>
+ */
+class MissingReadOnlyByPhpDocPropertyAssignRule implements Rule
+{
+
+	/** @var array<string, string[]> */
+	private array $additionalConstructorsCache = [];
+
+	/**
+	 * @param string[] $additionalConstructors
+	 */
+	public function __construct(
+		private array $additionalConstructors,
+	)
+	{
+	}
+
+	public function getNodeType(): string
+	{
+		return ClassPropertiesNode::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!$scope->isInClass()) {
+			throw new ShouldNotHappenException();
+		}
+		$classReflection = $scope->getClassReflection();
+		[$properties, $prematureAccess, $additionalAssigns] = $node->getUninitializedProperties($scope, $this->getConstructors($classReflection), []);
+
+		$errors = [];
+		foreach ($properties as $propertyName => $propertyNode) {
+			if (!$propertyNode->isReadOnlyByPhpDoc()) {
+				continue;
+			}
+			$errors[] = RuleErrorBuilder::message(sprintf(
+				'Class %s has an uninitialized readonly property $%s. Assign it in the constructor.',
```

Comment:
`readonly` -> `@readonly` - so that we have a slight difference and know it's from a PHPDoc :)